warehouse/templates/includes/packaging/project-data.html

```diff
@@ -43,12 +43,23 @@ <h6>{% trans %}Maintainers{% endtrans %}</h6>
       </span>
     {% endfor %}
   {% endif %}
+  {% if release.github_repo_info_url and release.github_open_issue_info_url and release.is_source_verified %}
+    <h6>{% trans %}Project links{% endtrans %}</h6>
+    {% set url = release.urls["Homepage"] %}
+    {% if is_valid_uri(url) %}
+    <li>
+      <a class="vertical-tabs__tab vertical-tabs__tab--with-icon vertical-tabs__tab--condensed" href="{{ url }}" rel="nofollow">
+        {{ url_icon("github.com", "github.com") }}Source <i class="fa fa-circle-check check"></i>
```

Comment:
I think we're mixing up 'Homepage' and 'Source' here -- these are two separate URLs, both of which may or may not be present. I would expect this to do something like below:

```
for name, url in release.urls.items()
    if name in {'Source', 'Homepage'} and is_valid_uri(url) and project.is_verified_url(url):
        <insert links here>
```

Reply:
Note that I think we need a generic function `is_verified_url` on the `Project`, not URL-specific functions on the `Release`, since verification should be shared across releases.